### DISCUSSION TITLE
Wishlist :scroll: of new features :cherries: in `dianna`

### DISCUSSION THREAD

**Discussion by geek-yang** [dianna-ai/dianna#619] [Ideas]
This is a forum for `dianna` users :computer: /enthusiasts :sparkling_heart: /developers :wrench: to share their wishes for new features of `dianna`.

In your post, you can make a wish and explain why you want the new feature. Please list ONLY 1️⃣  wish in each post following this format:

:bulb: Add support for xxx data
xxx data is always used in xxxx research and we want explainable in this domain.

Also, please vote for  other people's ideas.

**Comment by geek-yang** [comment]
:bulb: Support tabular data
Scientific-wise, many researchers use tabular data. Technically, it is a low-hanging fruit which would be easy to implement.

**Reply by elboyran** [reply]
> 💡 Support tabular data Scientific-wise, many researchers use tabular data. Technically, it is a low-hanging fruit which would be easy to implement.

I definitely support this, but might overlap with Ou's plans, in that case we should not duplicate.

**Reply by geek-yang** [reply]
Yes, I agree. I put this here mainly for an archive purpose 😉. I will come up with more ideas.

**Reply by elboyran** [reply]
Now we know, that there'll be no duplication if we include it.

**Reply by cpranav93** [reply]
Yes, I think Ou's use-case is in its current state not suitable for XAI application. So we can focus on tabular development independently and I think its a great idea.

**Comment by elboyran** [comment]
💡 More realistic earth observation use-case (dataset, model and XAI)
Select a dataset and pre-trained model from the [AiTLAS benchmarking suite](https://github.com/biasvariancelabs/aitlas-arena) (areal images for classification tasks) .

Foe example the [AiD dataset](https://www.kaggle.com/datasets/jiayuanchengala/aid-scene-classification-datasets), also part of the AiTLAS suite and analyze the XAI output as described for example in [this blog](https://blog.softwaremill.com/explainable-ai-xai-in-remote-sensing-classification-tasks-922d7720d88c) (but NOT with this naive model, rather the model from AiTLAS).

**Reply by cpranav93** [reply]
I like the idea but I would question why EO in particular? Why not another domain - especially since the tool is domain-agnostic?

**Reply by elboyran** [reply]
Only to have some concrete use case. And to relate it to your Section ;-) Any other is fine with me.

**Reply by cpranav93** [reply]
Its great to relate it to EO ;) Did not want to bias Dianna towards EO :P 

I am happy to give this a go especially if the data-set is readily available.

**Reply by geek-yang** [reply]
Here is a notebook on Kaggle showing a well-trained model for this task https://www.kaggle.com/code/gpiosenka/scenes-f1-score-94-5

**Comment by geek-yang** [comment]
:bulb: Provide XAI solutions for spatio-temporal data
In earth science/ remote sensing / environmental science domain, spatio-temporal data is very common. We could explore solutions for this by either (1) implementing new approaches (2) adapting our current method to it (e.g. flatten spatial dimensions and treat the data as timeseries).

**Reply by cpranav93** [reply]
As I mentioned above, it might be interesting to first analyse how diversely this spatio-temporal data is used across different domains to maintain the domain-agnostism (is that a word?) of Dianna

**Comment by geek-yang** [comment]
:bulb: Explore masking strategy to further improve the results
All the methods in dianna are perturbation based approaches and they rely a lot on the masking strategy. Especially for timeseries, there is huge potential for improving the results with smarter masking strategy.

**Reply by cwmeijer** [reply]
Related to this is the **option to use LIME's image masking strategy with RISE and vice versa**. For timeseries we built the code like that, but for images we don't support selecting masking strategies. My suspicion is that this tremendously different masking strategies are mostly the cause for any differences in the outcomes between LIME and RISE. I would like to know for sure.

**Reply by elboyran** [reply]
@geek-yang , @cwmeijer is Willem's recent PR solving this?

**Reply by cpranav93** [reply]
I really like the idea of a more efficient masking strategy. The lime segment paper had something related to this but I think that is already implemented, @geek-yang ? We can look further in to the literature for this and improve the methods.

**Comment by cwmeijer** [comment]
Write a tutorial for the FRB timeseries usecase #618.

It's by far the best working, most scientific and realistic (timeseries) use case that we have.

**Reply by cpranav93** [reply]
@cwmeijer , can I ask what the FRB usecase is? is it fast-radio bursts?

**Comment by elboyran** [comment]
💡 Add support for regression
We currently support only classification tasks, while regression tasks is often part of the research especially in climate sciences.

**Reply by cpranav93** [reply]
This would have a wide application use and I think we should definitely think about supporting this in Dianna

**Comment by cwmeijer** [comment]
Refactor RISE code over 3 files in accordance with our 3 data domains. #544.

For better maintainability.

**Comment by cwmeijer** [comment]
Video data format support including some way of viewing the result.

I would like to use this for Vocalisations of Disgust project for example.

**Comment by cpranav93** [comment]
Look into possible linguistic explanations along with the heat-maps? This seems to be picking up some interesting in the community. I can add provide some details/literature on this if we want to try it out. FYI, it does seem quite complicated to implement

**Comment by elboyran** [comment]
💡 Outreach

- Generic tutorial ->
- -> ONNX PR
- New movie with all modalities and the new dashboard
- use-case (e.g. EO) tutorials/demo
- material for internal/external training?
